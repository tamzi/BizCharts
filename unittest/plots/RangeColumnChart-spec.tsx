import React from 'react';
import RangeColumnChart from '../../src/plots/RangeColumnChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { type: '分类一', values: [76, 100] },
  { type: '分类二', values: [56, 108] },
  { type: '分类三', values: [38, 129] },
  { type: '分类四', values: [58, 155] },
  { type: '分类五', values: [45, 120] },
  { type: '分类六', values: [23, 99] },
  { type: '分类七', values: [18, 56] },
  { type: '分类八', values: [18, 34] },
];

describe('Plots-RangeColumnChart', () => {
  test('区间条形图-旧版', () => {
    let chart = null;
    render(<RangeColumnChart
      data={MOCK_DATA}
      title={{
        visible: true,
        text: '区间柱状图',
      }}
      yField='values'
      xField='type'
      color='l(90) 0:#3e5bdb 1:#b4d9e4'
      columnStyle={{
        fillOpacity: 0.8,
      }}
      label={{
        visible: true,
      }}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
