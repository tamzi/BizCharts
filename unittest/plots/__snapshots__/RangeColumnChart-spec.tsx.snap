// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-RangeColumnChart 区间条形图-旧版 1`] = `
Object {
  "animation": true,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "color": "l(90) 0:#3e5bdb 1:#b4d9e4",
  "columnStyle": Object {
    "fillOpacity": 0.8,
  },
  "columnWidthRatio": 0.6,
  "data": Array [
    Object {
      "type": "分类一",
      "values": Array [
        76,
        100,
      ],
    },
    Object {
      "type": "分类二",
      "values": Array [
        56,
        108,
      ],
    },
    Object {
      "type": "分类三",
      "values": Array [
        38,
        129,
      ],
    },
    Object {
      "type": "分类四",
      "values": Array [
        58,
        155,
      ],
    },
    Object {
      "type": "分类五",
      "values": Array [
        45,
        120,
      ],
    },
    Object {
      "type": "分类六",
      "values": Array [
        23,
        99,
      ],
    },
    Object {
      "type": "分类七",
      "values": Array [
        18,
        56,
      ],
    },
    Object {
      "type": "分类八",
      "values": Array [
        18,
        34,
      ],
    },
  ],
  "interactions": Array [
    Object {
      "type": "active-region",
    },
  ],
  "isRange": true,
  "label": Object {
    "visible": true,
  },
  "legend": false,
  "marginRatio": 0.03125,
  "renderer": "canvas",
  "tooltip": Object {
    "offset": 20,
    "shared": true,
    "showMarkers": false,
  },
  "xAxis": Object {
    "label": Object {
      "autoHide": Object {
        "cfg": Object {
          "minGap": 6,
        },
        "type": "equidistance",
      },
      "autoRotate": false,
    },
    "nice": true,
  },
  "xField": "type",
  "yAxis": Object {
    "label": Object {
      "autoHide": true,
      "autoRotate": false,
    },
    "nice": true,
  },
  "yField": "values",
}
`;
